overlord/snapstate/snapstate.go

```diff
@@ -579,11 +579,21 @@ func doInstall(st *state.State, snapst *SnapState, snapsup *SnapSetup, flags int
 	prev = setupSecurity
 
 	// finalize (wrappers+current symlink)
+	//
+	// For essential snaps that require reboots, 'link-snap' is currently marked as the edge of that reboot
+	// sequence. This means that we currently expect 'link-snap' to request the reboot and be the last task
+	// to run before the reboot takes place (for that lane/change). This task is assigned the edge 'MaybeRebootEdge'
+	// to indicate this.
```

Comment:
For link-snap, I assume that the reason for it to be the last one before the reboot is that at that point a reboot would use the new kernel but we do not want to proceed to do more things until we know that the kernel actually can boot, maybe it would be good to add this to the comment.

Reply:
Shortened the lines, also added a comment about this, thank you!